### PR TITLE
docs: Added documentation about "AlternateContent" feature

### PR DESCRIPTION
**Pull request by carldebilly**

### CODE REVIEW

File: doc/material-controls-extensions.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ``` xml
+    <ComboBox Style="{StaticResource MaterialComboBoxStyle}">
+        <um:ControlExtensions.Icon>
+            <SymbolIcon Symbol="SolidStar" />
+        </um:ControlExtensions.Icon>
+    </ComboBox>
+  ```
+
+## Alternate Content
+This feature allows to put a different content on a control when the state changes.
```
Comment (jeromelaban) [inline doc/material-controls-extensions.md:25]: What's the state property? Is it per control?

Comment (Xiaoy312) [reply doc/material-controls-extensions.md:25]: control specific, and only currently ToggleButton uses it.

Comment (agneszitte-nventive) [inline doc/material-controls-extensions.md:25]:
```suggestion
This feature allows putting different content on a control when the state changes.
It's control specific and for now, you can only use it with the ToggleButton control.
```

File: doc/material-controls-extensions.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Material Control Extensions
+
+## Icon
+This feature allows to put an icon on some supported control. Those icons could be any kind of _Icon_: `<BitmapIcon />`, `<FontIcon />`, `<SymbolIcon/>`, or `<PathIcon />`.
```
Comment (agneszitte-nventive) [inline doc/material-controls-extensions.md:4]:
```suggestion
This feature allows putting an icon on some supported control. Those icons could be any kind of _Icon_: `<BitmapIcon />`, `<FontIcon />`, `<SymbolIcon/>`, or `<PathIcon />`.

```

**Comment by agneszitte-nventive** [comment]
@carldebilly maybe you can double check the comments we left for this PR ;)

File: doc/material-controls-extensions.md (hunk 2 above)
Comment (Xiaoy312) [inline doc/material-controls-extensions.md:4]:
```suggestion
This feature allows for the addition of icon on the supported controls. Those icons could be any of the [`IconElement`](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.iconelement)s: `<BitmapIcon />`, `<FontIcon />`, `<PathIcon />` or `<SymbolIcon />`.
```

Comment (Xiaoy312) [reply doc/material-controls-extensions.md:4]: might want to clarify which icons are supported and on what platforms here
i have a feeling that we probably dont support all of them..?
perhaps we can expand the [IconElement doc](https://platform.uno/docs/articles/implemented/windows-ui-xaml-controls-iconelement.html) and link it here? or, it is already somewhere?

File: doc/material-controls-extensions.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+This feature allows to put a different content on a control when the state changes.
+
+### Alternate Content on ToggleButton
+``` xml
+    <ToggleButton Style="{StaticResource MaterialToggleButtonIconStyle}">
+        <!-- This is the default content - which is when the control state is UNCHECKED (the default value of a ToggleButton) -->
+        <PathIcon Data="{StaticResource Icon_more_horizontal}" />
+
+        <um:ControlExtensions.AlternateContent>
+        <!-- This is the alternate content - which is when the control state is CHECKED -->
```
Comment (Xiaoy312) [inline doc/material-controls-extensions.md:34]: indent alignment:
```suggestion
        <!-- This is the alternate content - which is when the control state is CHECKED -->
        <um:ControlExtensions.AlternateContent>
```

**Comment by agneszitte-nventive** [comment]
@carldebilly any update for this PR ;) ?

**Comment by agneszitte-nventive** [comment]
@carldebilly only missing the correct conventional commit for your last commit for this PR ;)

**Review by Xiaoy312 — APPROVED** [review]

**Pull request merged**